cmake/Platform.cmake

```diff
@@ -22,4 +22,34 @@ if (MSVC)
     # Suppress warnings: "Prefer enum class over enum" (Enum.3)
 
     add_compile_options(/wd4267 /wd26812)
+
+    if(NOT QL_MSVC_DISABLE_PARALLEL_BUILDS)
+        # enable parallel builds
+        add_compile_options(/MP)
+    endif()
+
+    # warning level 3 and all warnings as errors
+    add_compile_options(/W3 /WX)
```

Comment:
Not idiomatic.  Breaks debug builds.  Don't hard code these flags.  At worst, you _might_ define some baseline flags using `CMAKE_CXX_FLAGS_RELEASE` and `CMAKE_CXX_FLAGS_DEBUG` but even those I would omit.  Let the user be free to define what they need and not have to unwind hard-coded flags.